src/status_im/transport/inbox.cljs

```diff
@@ -136,69 +157,110 @@
-   peers or connects to it if it's not the case"
-  [{:keys [db] :as cofx}]
+  "There is only 2 summary changes that require offline inboxing action:
+  - mailserver disconnected: we try to reconnect
+  - mailserver connected: we mark the mailserver as trusted peer"
+  [previous-summary {:keys [db] :as cofx}]
   (when (and (:account/account db)
              config/offline-inbox-enabled?)
-    (let [{:keys [peers-summary peers-count]} db]
-      (if (zero? peers-count)
-        (update-mailserver-status :disconnected cofx)
-        (let [wnode (get-current-wnode-address db)]
-          (if (registered-peer? peers-summary wnode)
-            (handlers-macro/merge-fx cofx
-                                     {::mark-trusted-peer {:web3  (:web3 db)
-                                                           :wnode wnode}}
-                                     (generate-mailserver-symkey))
-            (connect-to-mailserver cofx)))))))
-
-(defn get-topics
-  [db topics discover?]
-  (let [inbox-topics    (:inbox/topics db)
-        discovery-topic (transport.utils/get-topic constants/contact-discovery)
-        topics          (or topics
-                            (map #(:topic %) (vals (:transport/chats db))))]
-    (cond-> (apply conj inbox-topics topics)
-      discover? (conj discovery-topic))))
+    (let [{:keys [peers-summary peers-count]} db
+          wnode                               (get-current-wnode-address db)
+          mailserver-was-registered?          (registered-peer? previous-summary
+                                                                wnode)
+          mailserver-is-registered?           (registered-peer? peers-summary
+                                                                wnode)
+          mailserver-connected?               (and mailserver-is-registered?
+                                                   (not mailserver-was-registered?))
+          mailserver-disconnected?            (and mailserver-was-registered?
+                                                   (not mailserver-is-registered?))]
+      (cond
+        mailserver-disconnected?
+        (connect-to-mailserver cofx)
+
+        mailserver-connected?
+        {::mark-trusted-peer {:web3  (:web3 db)
+                              :wnode wnode}}))))
+
+(defn inbox-ready? [{:keys [db]}]
+  (let [mailserver-status (:mailserver-status db)
+        sym-key-id        (:inbox/sym-key-id db)]
+    (and (= :connected mailserver-status)
+         sym-key-id)))
+
+(defn get-request-messages-topics
+  "Returns topics for which full history has already been recovered"
+  [db]
+  (conj (mapv :topic
+              (remove :fetch-history?
+                      (vals (:transport/chats db))))
+        (transport.utils/get-topic constants/contact-discovery)))
+
+(defn get-request-history-topics
+  "Returns topics for which full history has not been recovered"
+  [db]
+  (mapv :topic
```

Comment:
is there a reason to use `mapv`, but not `map` here? Code is similar to the above in `get-request-messages-topics` fn where it really needs vector to `conj`  with

Reply:
it's just for consistency, lazyness doesn't matter here anyway